pkg/ui/src/TextEditable/TextEditable.svelte

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+	import { Pencil, Check, X } from 'lucide-svelte';
+
+	/**
+	 * This is the exposed 'value' of the text/input element. This value accepts updates from the parent component, while
+	 * the internal, display value is updated when the input is saved. This results in the following behaviour:
+	 * - parent component binds to this value
+	 * - the parent component updates the 'value' and the udpate is immediately visible in the input
+	 * - the user edits the input - no change is propagated to the parent component
+	 * - the user saves the input - the parent component is updated with the new value
+	 */
+	export let value = '';
+
+	/** This is the internal value, used to store the current state of the input */
+	$: text = value;
+	export let isEditing = false;
+
+	console.log('Is editing: ' + isEditing);
+
+	/** Enter edit mode */
+	function edit() {
+		isEditing = true;
```

Comment:
An alternative to `autofocus` would be:
1. Bind to an inputRef `bind:this={inputRef}`
2. After toggling `isEditing = true` , add `tick().then(() => (inputRef as HTMLElement).focus())`

Reply:
👍